en/scripting/modules/example.md

```diff
@@ -0,0 +1,248 @@
+    const original = fs.readFileSync(file, { encoding: 'utf8' });
+    fs.writeFileSync(file, `
+    namespace proto {
+        ${original}
+    }
+    export default proto;
+    `);
+    ```
+
+    Replace the original ``build-proto:pbts`` command with the following:
+
+    ```json
+    "build-proto:pbts": "pbts --main --out ./Proto.js/proto.d.ts ./Proto.js/*.js && node ./Tools/wrap-pbts-result.js"
+    ```
+
+Eventually, it will be ready to run directly. For the full project content, please refer to: [npm-case](https://github.com/cocos-creator/example-3d/tree/master/npm-case).
+
+> **Note**: the packaged js file can be placed in the `assets` directory of the project or elsewhere in the project. The js file in the `assets` directory no longer needs to be checked for export as a plugin, please be aware of that.
+
+### Case 3: use of lodash-es
+
+Similar to [Case 1: protobufjs use](# Case 1: protobufjs-use), install the `lodash-es` package. We know that the entry file is `lodash.js`, and the entry file also automatically helps to export all submodules under it in **ESM** module format, and according to `type` it also confirms that it is currently an **ESM** module. Therefore, it is possible to import any module directly. Let's take the `test.ts` script asset under `assets` as an example, and introduce the submodules within `lodash`.
+
+```ts
+import { array, add } from 'lodash-es';
+```
+
+At this point, you will find that an error is reported at the code level, but it actually works. This is because there is a clear distinction between JavaScript, which is dynamically typed, and TypeScript, which is statically typed, so there is no way to know the exact type of the exported module when using js scripts, and the best thing to do is to declare a type definition file .d.ts. Fortunately, but when we hover over the error, there is a hint The type declaration file for the `lodash` module can be installed by running `npm i --save-dev @types/lodash-es`. After installation, restart VS Code and you will see that the error message disappears, along with the code hint.
+
+### Case 4: Using web3
+
+Install the `web3` package in the same way as the `protobufjs` case above. Using the same detection method, determine that `web3` is a **CJS** module. Import in the same way.
+
+```ts
+import web3 from 'web3';
+```
+
+When I go back to the editor, I still get a bunch of errors as follows.
+
+![node-module](./example/node-module.png)
+
+This is because the package is customized for Node and contains built-in modules from Node.js, which are not available in Creator, resulting in errors. Usually these packages also take care of web users. When we import the package directly, the package entry points directly to the Node version, while the web version is placed in the `dist/**.js` or `dist/**.min.js` file under the package. In this case, you will need to import the custom version of the Web provided in the package as follows
```

Comment:
```suggestion
This is because the package is customized for Node and contains built-in modules from Node.js, which are not available in Creator, resulting in errors. Usually, these packages also take care of web users. When importing the package directly, the package entry points directly to the Node version, while the web version is placed in the `dist/**.js` or `dist/**.min.js` file under the package. In this case, it is necessary to import the custom version of the Web provided in the package as follows:
```